### ISSUE TITLE
Evaluation of array.filter with function pointer not working

### ISSUE THREAD

**Issue by SvenMitlehner** [scriban/scriban#320]
Using the exampe template from the documentation of array.filter:

`var result = Template.Parse("{{['', '200', '','400'] | array.filter @string.empty}}").Replace(new TemplateContext());`

The above gives the desired result: 

`["", ""]`

Using Evaluate instead of Replace:

`var result = Template.Parse("{{['', '200', '','400'] | array.filter @string.empty}}").Evaluate(new TemplateContext());`

returns an object of type ScribanRange. When trying to iterate the items in result, i get the following exception:

 ```
Value cannot be null. (Parameter 'callerContext')
   at Scriban.Syntax.ScriptFunctionCall.Call(TemplateContext context, ScriptNode callerContext, IScriptCustomFunction function, ScriptArray arguments)
   at Scriban.Functions.ArrayFunctions.FilterInternal(TemplateContext context, SourceSpan span, IEnumerable list, IScriptCustomFunction function, Type destType)+MoveNext()
   at Scriban.Runtime.ScriptRange.GetEnumerator()+MoveNext()
```

Expectation would be to either be able to iterate over the result or directly getting an array or list as a result. Am I doing anything wrong here?

**Comment by NeilMacMullen** [comment]
I don't think it is related to "filter" specifically.  You can seem similar results with array.each....

```
var result = Template
                .Parse("{{[1,2,3] | array.each @(do;ret $0;end) }}")
                .Evaluate(new TemplateContext());
```

**Comment by xoofx** [comment]
There is a bug in `ScriptFunctionCall.Call`, likely easily fixable

**Comment by n1l** [comment]
@xoofx https://github.com/scriban/scriban/pull/321

**Comment by NeilMacMullen** [comment]
@n1l   Interesting - from your PR description I wonder if this is also causing this kind of error...

```
{{

#test that a parameter can be captured by a local function declaration

func t(fn)
  ret fn
end

func call_func(p)
  ret  t @(do;ret p;end)
end

func call_array(p)
  ret  [1,2,3] | array.each @(do;ret p;end)
end

#This succeeds: parameter is accessible to local function when calling into another function
call_func true 


#This fails: parameter is not found at line 14 when calling into array iterator
# "Render error: <input>(14,38) : error : The variable or function `p` was not found"
call_array true

}}
```

**Comment by NeilMacMullen** [comment]
Indeed, the two issues appear to have the same root cause - see #322

**Comment by xoofx** [comment]
Fixed by commit d77a23736ecc3a90fe204799ee53e73a929e5870

**Comment by NeilMacMullen** [comment]
nice - thanks!
